frontend/include/chpl/resolution/resolution-types.h

```diff
@@ -1956,6 +1996,208 @@ class CallResolutionResult {
   /// \endcond DO_NOT_DOCUMENT
 };
 
+/* The result of resolving a 'these' call. This contains more information
+   than a general call resolution result, because we can encode what threw off
+   the iteration (e.g., lack of a follower iterator)
+
+   In the case of failure due to zippering (e.g., we're iterating over a
+   loop expression, but the Nth zippred iterand doesn't have the appropriate
+   iterator), this type contains a 'zipperedFailure_' field, which contains
+   the result of trying to iterate the Nth iterand, and a 'zipperedFailureIndex_'
+   which is the index of the failed iterand.
+*/
+struct TheseResolutionResult {
+ public:
+  enum FailureReason {
+    /* We didn't fail, finding an iterator. */
+    THESE_SUCCESS = 0,
+    /* We didn't attempt to resolve this iterator, possibly because it's
+       not supported in this context. */
+    THESE_NOT_ATTEMPTED,
+    /* We tried to resolve a standalone iterator on a loop expression, but
+       loop expressions don't have standalone iterators. */
+    THESE_FAIL_NO_LOOP_EXPR_STANDALONE,
+    /* We tried to resolve a standalone iterator on a promoted expression, but
+       promoted expressions don't have standalone iterators. */
+    THESE_FAIL_NO_PROMO_STANDALONE,
+    /* We tried to resolve a parallel iterator by we're iterating over a serial
+       loop expression. */
+    THESE_FAIL_SERIAL_LOOP_EXPR,
+    /* We couldn't find the given iterator's definition. */
+    THESE_FAIL_NO_ITERATOR_WITH_TAG,
+    /* We found a leader and a follower, but the leader's yield type doesn't match
+       the follower's 'followThis'. */
+    THESE_FAIL_LEADER_FOLLOWER_MISMATCH,
+    /* We tried to resolve a zippered-like iterator (zippered loop expression,
+       promoted expressions) and one of the iterables that makes it up failed
+       to resolve. In this case, the zipperedFailure_ and zipperedFailureIndex_
+       fields are set. */
+    THESE_FAIL_ZIPPERED_ARG_FAILED,
+    /* We started resolving a zippered-like iterator using (e.g.), a leader
+       iterator, so we didn't attempt to use a follower iterator. */
```

Comment:
This reflects an error that the Dyno implementation is currently producing in which we "commit" to a leader iterator if we find a leader iterator on the first iterand, even if it's a serial-capable loop, we only resolve followers subsequently. This means an iteration that _could_ work with `for` doesn't work with a bracket loop if the first iterand has a leader, but the others don't.

I double checked and this doesn't occur in production. I'm going to note this as future work, because I don't want to pile onto this PR